src/ts/navigation/navigation.ts

```diff
@@ -0,0 +1,320 @@
+  toggleWrapper.classList.add('cads-greedy-nav');
+
+  const dropdownId = 'cads-greedy-nav-dropdown';
+
+  toggleWrapper.insertAdjacentHTML(
+    'beforeend',
+    buildToggle(containerEl, dropdownId),
+  );
+
+  toggleWrapper.appendChild(buildNavDropdown(dropdownId));
+
+  mainNavEl.insertAdjacentElement('afterend', toggleWrapper);
+}
+
+function setToggleVisibility(containerEl: HTMLElement, breaks: number[]) {
+  const toggleEl = getToggleEl(containerEl);
+
+  if (breaks.length < 1) {
+    toggleEl.parentElement?.setAttribute('aria-haspopup', 'false');
+  } else {
+    toggleEl.parentElement?.setAttribute('aria-haspopup', 'true');
+  }
+}
+
+function openDropDown(containerEl: HTMLElement) {
+  const data = extractDataAttributes(containerEl);
+  const toggleEl = getToggleEl(containerEl);
+  const dropdownEl = getDropdownEl(containerEl);
+
+  toggleEl.setAttribute('aria-expanded', 'true');
+  dropdownEl.setAttribute('aria-hidden', 'false');
+
+  toggleEl.firstElementChild!.textContent = data.labelClose;
+  toggleEl.setAttribute('aria-label', data.ariaLabelClose);
+}
+
+function closeDropDown(containerEl: HTMLElement) {
+  const data = extractDataAttributes(containerEl);
+  const toggleEl = getToggleEl(containerEl);
+  const dropdownEl = getDropdownEl(containerEl);
+
+  toggleEl.setAttribute('aria-expanded', 'false');
+  dropdownEl.setAttribute('aria-hidden', 'true');
+
+  toggleEl.firstElementChild!.textContent = data.label;
+  toggleEl.setAttribute('aria-label', data.ariaLabel);
+}
+
+function toDropdown(containerEl: HTMLElement) {
+  const navDropdown = getDropdownEl(containerEl);
+  const mainNav = getMainNavEl(containerEl);
+
+  if (mainNav && mainNav.children.length > 0 && mainNav.lastElementChild) {
+    if (navDropdown.firstChild) {
+      navDropdown.insertBefore(
+        mainNav.lastElementChild,
+        navDropdown.firstChild,
+      );
+    } else {
+      navDropdown.appendChild(mainNav.lastElementChild);
+    }
+  }
+}
+
+function toMenu(containerEl: HTMLElement) {
+  const navDropdown = getDropdownEl(containerEl);
+  const mainNav = getMainNavEl(containerEl);
+
+  if (
+    mainNav &&
+    navDropdown &&
+    navDropdown.children.length > 0 &&
+    navDropdown.firstElementChild
+  ) {
+    mainNav.appendChild(navDropdown.firstElementChild);
+  }
+}
+
+function computeContentWidth(containerEl: HTMLElement) {
+  const styles = window.getComputedStyle(containerEl);
+  const padding =
+    parseFloat(styles.paddingLeft) + parseFloat(styles.paddingRight);
+
+  return containerEl.clientWidth - padding;
+}
+
+function computeChildrenOffsetWidth(containerEl: HTMLElement) {
+  const children = containerEl.childNodes;
+
+  let sum = 0;
+  for (let i = 0; i < children.length; i++) {
+    const child = <HTMLElement>children[i];
+
+    if (child.nodeType !== 3 && !Number.isNaN(child.offsetWidth)) {
+      sum += child.offsetWidth;
+    }
+  }
+
+  return sum;
+}
+
+function doesItFit(containerEl: HTMLElement, breaks: number[]) {
```

Comment:
What is breaks here?  I think it has something to do with the cumulative widths of each nav link but it would be good to make it explicit (in a comment would be fine I think).

In fact some documentation for this function would be good - it's hard to infer the behaviour from the code fully.

Also it seems to do more than answer the question 'does it fit' - should it be broken up or renamed?

Reply:
Good shout.

Short answer: Agreed, let's break this down more. 

Longer answer: For context the the process I followed was basically to copy over the existing file and names and gradually refactor the implementation, rather than a ground-up rewrite. I'd bet on this being part that retains the most naming of the original.

Reply:
OK - I'm happy for it to stay like this then and maybe this can be a smaller refactor later down the line.  It would be good to have another look at it at some point tho.